thrimshim/thrimshim/main.py

```diff
@@ -36,6 +40,53 @@ def _start_response(status, headers, exc_info=None):
 	return handle
 
 
+
+def authenticate(f):
+	"""Authenticate a token against the database.
+
+	Reference: https://developers.google.com/identity/sign-in/web/backend-auth"""
+	@wraps(f)
+	def decorated_function(*args, **kwargs):
+		if flask.request.method == 'POST':
+			if app.no_authentication:
+				return f(*args, editor='NOT_AUTH', **kwargs)
+
+			userToken = flask.request.json['token']
```

Comment:
A missing token should return a 402 (auth required), rather than erroring. A 400 is also acceptable (which might actually be what Flask does for you here. I know it does that for missing values in `request.params`, but I'm not sure about `request.json`)

Reply:
401 is Unauthorized. 402 is Payment Required.

Reply:
right, derp.